apps/api-extractor/src/generators/ApiReportGenerator.ts

```diff
@@ -426,15 +496,50 @@ export class ApiReportGenerator {
           }
         }
 
-        ApiReportGenerator._modifySpan(collector, child, entity, childAstDeclaration, insideTypeLiteral);
+        ApiReportGenerator._modifySpan(
+          collector,
+          child,
+          entity,
+          childAstDeclaration,
+          insideTypeLiteral,
+          releaseLevel
+        );
       }
     }
   }
 
-  private static _shouldIncludeInReport(astDeclaration: AstDeclaration): boolean {
+  private static _shouldIncludeInReport(
+    collector: Collector,
+    astDeclaration: AstDeclaration,
+    releaseLevel: ApiReportReleaseLevel
+  ): boolean {
     // Private declarations are not included in the API report
     // eslint-disable-next-line no-bitwise
-    return (astDeclaration.modifierFlags & ts.ModifierFlags.Private) === 0;
+    if ((astDeclaration.modifierFlags & ts.ModifierFlags.Private) !== 0) {
+      return false;
+    }
+
+    const apiItemMetadata: ApiItemMetadata = collector.fetchApiItemMetadata(astDeclaration);
+
+    // No specified release tag is considered the same as `@public`.
+    const releaseTag: ReleaseTag =
+      apiItemMetadata.effectiveReleaseTag === ReleaseTag.None
+        ? ReleaseTag.Public
+        : apiItemMetadata.effectiveReleaseTag;
+
+    // If the declaration has a release tag that is not in scope, omit it from the report.
+    switch (releaseLevel) {
+      case ApiReportReleaseLevel.Untrimmed:
+        return true;
+      case ApiReportReleaseLevel.Alpha:
+        return releaseTag >= ReleaseTag.Alpha;
+      case ApiReportReleaseLevel.Beta:
+        return releaseTag >= ReleaseTag.Beta;
```

Comment:
Design question: Is it beneficial to duplicate the `@public` signatures in the `.beta.api.md` and `.alpha.md` files? From an approval standpoint, if I already approved the `.public.api.md` file then why do I need to reexamine the exact same API's in two other files?  Unlike a .d.ts rollup, there is no technical requirement for the .api.md report to include all of its dependent types.  Actually its main design goal is to avoid triggering spurious CODEOWNERS approvals.

Reply:
It's an interesting semantics question. I was operating with the following goals in mind:
1. To know what appears in a given API release level, you can look at a single report.
2. Each report mirrors its corresponding trimmed rollup, contents-wise.

I think these are useful invariants, but they do come at the cost of some duplication. We could also make this configurable - allow users to specify if they want all API items that would be included at the release level XOR only include items with the specific tag.

As for the design goal, this pattern still addresses the needs, because we can put different (looser) requirements for changes to `beta.api.md` than `public.api.md`.

Reply:
That makes sense. And your "variants" design leaves open the possibility of introducing `beta-only` or `public-only` variants in the future if someone wants that, right?

Reply:
Yeah, we can add whatever flexibility we want by adding new "enum" options and adding handler logic for them.